### ISSUE TITLE
Emit error on space characters in alias definitions before/after equals sign

### ISSUE THREAD

**Issue by WalterGR** [koalaman/shellcheck#2442]
#### For new checks and feature suggestions
- [*] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [*] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash

alias space_before_equals ='echo first alias'
alias space_after_equals= 'echo second alias'

```

#### Here's what shellcheck currently says:

\<nothing\>

#### Here's what I wanted or expected to see:

An error listed for each alias definition.

Executing the script above results in the following bash errors:

```
> echo "$BASH_VERSION"
5.0.17(1)-release

> ./test.sh
/Users/walter/test.sh: line 3: alias: space_before_equals: not found
/Users/walter/test.sh: line 3: alias: =echo first alias: not found
/Users/walter/test.sh: line 4: alias: echo second alias: not found

>
```

#### Remarks:

`help alias` in bash doesn't explicitly mention that spaces are forbidden before and after the equals sign, but if you assume that `name[=value]` is literal, then it does implicitly mention it...

```
> echo "$BASH_VERSION"

> help alias
alias: alias [-p] [name[=value] ... ]
    Define or display aliases.
    
    Without arguments, `alias' prints the list of aliases in the reusable
    form `alias NAME=VALUE' on standard output.
    
    Otherwise, an alias is defined for each NAME whose VALUE is given.
    A trailing space in VALUE causes the next word to be checked for
    alias substitution when the alias is expanded.
    
    Options:
      -p	print all defined aliases in a reusable format
    
    Exit Status:
    alias returns true unless a NAME is supplied for which no alias has been
    defined.

>
```

**Comment by WalterGR** [comment]
This is relevant to the above.  https://unix.stackexchange.com/questions/410651/why-are-spaces-not-allowed-in-some-bash-commands-like-alias/410654#410654

> Q: Why are spaces not allowed in some bash commands, like 'alias'?
> 
> A: Because `alias` can be used to query aliases, and to set multiple aliases at one time:
> 
> ```sh
> $ alias foo                       # query for one alias
> bash: alias: foo: not found
> 
> $ alias foo = bar                 # query for three aliaes
> bash: alias: foo: not found
> bash: alias: =: not found
> bash: alias: bar: not found
> 
> $ alias foo=bar doo=dar           # set two aliases
> $ alias doo                       # query one alias
> alias doo='dar'
> 
> $ alias foo = bar                 # query three, again
> alias foo='bar'
> bash: alias: =: not found
> bash: alias: bar: not found
> ```
> 
> (Don't ask me why it says that the alias `=` is not found, instead of telling it's an invalid name for an alias. Which I think it is, I can't find a way to set an alias of that name.)

(Answer written by https://unix.stackexchange.com/users/170373/ilkkachu)

So while the answer is true ("alias` can be used to query aliases, and to set multiple aliases at one time") I don't see a scenario in which an alias with an equals sign with whitespace before or after can be legal bash.

**Comment by koalaman** [comment]
There was already a check that handles this exact case for `export`, `declare`, and such. I added `alias` to the list.

Thanks!

**Comment by WalterGR** [comment]
WOW what amazing turnaround!  Thank you so much!!
